### PR TITLE
Add input-tensor param to all KerasCV.models

### LINKED ISSUE #638: Re-introduce `input_tensor` in the keras_cv.models codebase

### PR DESCRIPTION
**Pull request by ianjjohnson**
Fixes #638

This is necessary for high performance of KerasCV models used as backbones in RetinaNet and other applications

@LukeWood

### CODE REVIEW

File: keras_cv/models/csp_darknet.py
Comment (LukeWood) [inline keras_cv/models/csp_darknet.py:115]: looks like we can use a utility function for this.  

Let's do:
keras_cv/models/utils.py

```
def parse_model_input(input_shape, input_tensor):
    if input_tensor is None:
       ....
    # return based on these values
```

File: keras_cv/models/csp_darknet.py
Comment (LukeWood) [inline keras_cv/models/csp_darknet.py:40]: lets add a unit test to models_test.py to cover this and make sure we can construct a model (like the backbone) using this API.

**Review by LukeWood — CHANGES_REQUESTED** [review]
minor comments, otherwise LGTM.

Thanks for the quick PR!

File: keras_cv/models/csp_darknet.py
Comment (ianjjohnson) [reply keras_cv/models/csp_darknet.py:41]: Done -- I've added a test and verified that without this new parameter, the test fails.

File: keras_cv/models/utils.py
Comment (LukeWood) [inline keras_cv/models/utils.py:21]: last thing we could do here is add test coverage over parse_model_inputs in `utils_test.py`

test:
- it constructs a keras.Input for input_shape
- it returns the tensor of the input_tensor is a keras tensor

Then LGTM, sorry for the late batch of comments.  Just thought of this after seeing the helper.

**Review by LukeWood — CHANGES_REQUESTED** [review]
added a comment, sorry about that!

File: keras_cv/models/utils.py
Comment (ianjjohnson) [reply keras_cv/models/utils.py:21]: Added a test case using assertIs to make sure the input tensor is returned when possible, and otherwise the shape matches up correctly

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thanks @ianjjohnson - looks like a merge conflict is blocking merge

**Comment by ianjjohnson** [comment]
> Thanks @ianjjohnson - looks like a merge conflict is blocking merge

I've fixed the merge conflict. Once CI passes I'll merge.

**Pull request merged**
